scope_registry.go

```diff
@@ -21,64 +21,90 @@
 package tally
 
 import (
+	"hash/maphash"
+	"runtime"
 	"sync"
 	"unsafe"
 )
 
 var scopeRegistryKey = keyForPrefixedStringMaps
 
 type scopeRegistry struct {
-	mu        sync.RWMutex
-	root      *scope
-	subscopes map[string]*scope
+	seed maphash.Seed
+	root *scope
+	// We need a subscope per GOPROC so that we can take advantage of all the cpu available to the application.
+	subscopes []*bucketScope
+}
+
+type bucketScope struct {
+	mu sync.RWMutex
+	s  map[string]*scope
 }
 
 func newScopeRegistry(root *scope) *scopeRegistry {
+	procs := runtime.GOMAXPROCS(-1)
```

Comment:
could we wire this in as part of ScopeOptions and config?
With GOMAXPROCs as default.